coordinator/src/services/translation/request/payload/factory.ts

```diff
@@ -0,0 +1,203 @@
+import Hapi from "@hapi/hapi"
+import {fhir, hl7V3} from "@models"
+import pino from "pino"
+import * as uuid from "uuid"
+import {
+  getMessageIdFromBundle,
+  getMessageIdFromClaim,
+  getMessageIdFromTask,
+  identifyMessageType
+} from "../../common"
+import {convertCancellation} from "../cancel/cancellation"
+import {convertDispenseClaim} from "../dispense/dispense-claim"
+import {convertDispenseNotification} from "../dispense/dispense-notification"
+import {translateReleaseRequest} from "../dispense/release"
+import {convertParentPrescription} from "../prescribe/parent-prescription"
+import {convertTaskToDispenseProposalReturn} from "../return/return"
+import {createSendMessagePayload} from "../send-message-payload"
+import {convertTaskToEtpWithdraw} from "../withdraw/withdraw"
+
+type BundleTranslationResult = hl7V3.ParentPrescriptionRoot
+  | hl7V3.CancellationRequestRoot
+  | hl7V3.DispenseNotificationRoot
+  | hl7V3.DispenseClaimRoot
+
+type ParametersTranslationResult = hl7V3.NominatedPrescriptionReleaseRequestWrapper
+  | hl7V3.PatientPrescriptionReleaseRequestWrapper
+
+type TaskTranslationResult = hl7V3.DispenseProposalReturnRoot | hl7V3.EtpWithdrawRoot
+
+type ClaimTranslationResult = hl7V3.DispenseClaimRoot
+
+type PayloadContent = BundleTranslationResult
+  | TaskTranslationResult
+  | ParametersTranslationResult
+  | ClaimTranslationResult
+
+type Payload<T extends PayloadContent> = {
+  content: T,
+  interactionId: hl7V3.Hl7InteractionIdentifier
+}
+
+type FactoryInput = fhir.Bundle | fhir.Task | fhir.Parameters | fhir.Claim
+
+export abstract class PayloadFactory {
+  protected abstract create(
```

Comment:
`create` is only used within the `PayloadFactory` class, so I don't think that would be the case?

Reply:
What I meant is that the method is not used anywhere else, aside from the unexported classes within the same module, so I didn't think it could cause any confusion. Anyway, since that's not the case, I've renamed it to `createPayload`, so the meaning should be clear now